tests/unit/s2n_build_test.c

```diff
@@ -65,6 +65,33 @@ S2N_RESULT s2n_test_lowercase_copy(const char *input, char *destination, size_t
     return S2N_RESULT_OK;
 }
 
+S2N_RESULT s2n_check_supported_libcrypto(const char *s2n_libcrypto)
+{
+    RESULT_ENSURE_REF(s2n_libcrypto);
+
+    /* List of supported libcrypto variants we test with */
+    const char *supported_libcrypto[] = {
+        "awslc",
+        "awslc-fips",
+        "awslc-fips-2022",
+        "boringssl",
+        "libressl",
+        "openssl-1.0.2",
+        "openssl-1.0.2-fips",
+        "openssl-1.1.1",
+        "openssl-3.0"
+    };
+
+    for (size_t i = 0; i < s2n_array_len(supported_libcrypto); i++) {
+        if (strcmp(s2n_libcrypto, supported_libcrypto[i]) == 0) {
+            return S2N_RESULT_OK;
+        }
+    }
```

Comment:
As discussed offline: I like it! But can we work s2n_libcrypto_is_openssl in? Even with the comment, I worry it's not clear enough that the real goal of this test is s2n_libcrypto_is_openssl